### PR TITLE
Fix youtube videos

### LINKED ISSUE #162: Links to Videos are broken
I have noticed several issues with videos : 

- I have a 500 when I visit https://paris-rb.org/videos/how-music-works-using-ruby-english

- When I visit this one https://paris-rb.org/videos/optimisation-tcp-le-clash, I have this error message
Capture d’écran 2023-01-03 à 2 42 43 PM

### PR DESCRIPTION
**Pull request by adrienpoly**
This PR includes #165 and completely closes #162

Some videos are provided with Vimeos url some are with Youtube.
I created some basic helper function in the Video model to provide a more generic `provider` and `provider_id` with it's respective partial to display the iframe

further enhancement could be to  create a `provider` enum on the Video model but not knowing exactly the content of the db I didn't wanted to push this fare yet

Open to discussion but this PR as it touch a view, starts converting SLIM to ERB.

Expected benefits from ERB is to stick as close as possible to Vanilla Rails

### CODE REVIEW

**Review by btrd — APPROVED** [review]

File: app/views/videos/_vimeo.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<iframe src="//player.vimeo.com/video/#{video.vimeo_id}"
```
Comment (sunny) [inline app/views/videos/_vimeo.html.erb:1]:
```suggestion
<iframe src="//player.vimeo.com/video/<%= video.provider_id %>" 
```

File: app/views/videos/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      <div class="col-md-8 col-md-offset-2">
+        <div class="embed-container">
+          <%= render "videos/#{video.provider}", video: video if video.provider %>
+        </div>
+      </div>
+    </div>
+
+    <div class="row">
+      <div class="video-description col-md-6 col-md-offset-3">
+        <p><%= simple_format(video.description) %></p>
```
Comment (sunny) [inline app/views/videos/show.html.erb:22]: Simple format already inserts paragraphs, so they should be removed here:

```suggestion
        <%= simple_format video.description %>
```

File: app/models/video.rb [hunk 3]
```diff
@@ -21,11 +21,37 @@ class Video < ApplicationRecord
   before_save :set_slug
   after_save :refresh_sitemap
 
+  def provider
+    return :vimeo if vimeo?  
+    return :youtube if youtube?
+  end
+
+  def provider_id
+    send("#{provider}_id") if provider
```
Comment (Intrepidd) [inline app/models/video.rb:30]: Safer to go through `attributes["#{provider}_id"]` no ?

Comment (Intrepidd) [reply app/models/video.rb:30]: Ah nvm just realised the `youtube_id` is a method, I'd still advocate against using `send` and rather use a ternary or a config hash

Comment (adrienpoly) [reply app/models/video.rb]: indeed

**Review by Intrepidd — APPROVED** [review]
LGTM :)

**Pull request merged**